pkg/utils/utils.go

```diff
@@ -32,3 +36,26 @@ func Contains(s []string, e string) bool {
 	}
 	return false
 }
+
+// PollUntilConditionIsMet validates if a certain condition is met at defined poll intervals.
+// If a timeout is reached, an associated error is returned to the caller.
+// meetsCondition contains the use-case specific code that returns true when a certain condition is achieved.
+func PollUntilConditionIsMet(pollInterval, timeOut <-chan time.Time, meetsCondition func() (bool, error)) error {
+	for done := false; !done; {
+		select {
+		case <-timeOut:
+			return fmt.Errorf("timed out while waiting for job to complete")
+		case <-pollInterval:
+			var done bool
+			var err error
+			if done, err = meetsCondition(); err != nil {
+				done = true
+				return err
+			}
+			if done {
+				return nil
+			}
+		}
```

Comment:
variable with same name may create confusion here, lets use a different name here

Reply:
Identified that `done` was redundant, remove the other `done` defined in the loop and simplified the logic further. 
Thank you!